lib/src/sources/cpuid_source.cc

```diff
@@ -7,10 +7,15 @@ namespace whereami { namespace sources {
 
     cpuid_registers cpuid_base::read_cpuid(unsigned int leaf, unsigned int subleaf) const {
         cpuid_registers result;
-#if defined(__x86_64__) || defined(__i386__)
+#if defined(__i386__)
         // ebx is the PIC register in 32-bit environments; Don't clobber it
         asm volatile(
-            "xchgl %%ebx,%k1; xor %%ebx,%%ebx; cpuid; xchgl %%ebx,%k1"
```

Comment:
One question I had: what was the xor doing?

Reply:
That part zeroes out the contents of `ebx`